x-pack/plugin/build.gradle

```diff
@@ -40,14 +40,6 @@ subprojects {
         // filter out non-existent classes directories from empty source sets
         final FileCollection classDirectories = project.files(files).filter { it.exists() }
 
-        onlyIf {
-          /*
-           * The latest version of ASM does not understand JDK 14. However, Gradle can not distinguish between JDK 13 and JDK 14 (treating
-           * anything above JDK 12 as JDK 13). So, to exclude JDK 14 until a newer version of ASM is available, we also have to exclude JDK
-           * 13. See https://github.com/elastic/elasticsearch/issues/45927.
-           */
-          Integer.parseInt(BuildParams.runtimeJavaVersion.majorVersion) < 13
```

Comment:
Should we keep this in here set to the latest compatible version anyhow? We are just going to likely have to put it back once we start testing with JDK 15 EA releases.

Reply:
FYI, if we do, this can be simplified simply to `BuildParams.runtimeJavaVersion < JavaVersion.VERSION_13`

Reply:
Oh, one more thing, why do we seem to insist on using the _runtime_ JDK for this stuff? I've already ripped this out of a bunch of places because this is a _build_ time check. I should not be using some old JDK for doing precommit checks just because I happen to have `RUNTIME_JAVA_HOME` set. I think this could cause wacky behavior for folks.

I'd prefer we use the `compilerJavaHome` here instead. I think it makes the most sense for these use cases.

Reply:
I pushed 90dd3ae6c4e27b144d6aec69565e9b3b9a84c190. This means that we don't need to keep the `onlyIf`, since we'll always be using a compatible compiler.